File: frontend/el-buen-sabor-frontend/src/app/entidades/DetalleFactura.ts
import { ArticuloInsumo } from './ArticuloInsumo';
import { ArticuloManufacturado } from './ArticuloManufacturado';
import { Factura } from './Factura';

export class DetalleFactura {
  id: number = 0;
  cantidad: number = 0;
  subtotal: number = 0;
  factura: Factura = new Factura();
  articuloManufacturado: ArticuloManufacturado = new ArticuloManufacturado();
  articuloInsumo: ArticuloInsumo = new ArticuloInsumo();
}

File: backend/src/main/java/com/elbuensabor/entities/DetalleFactura.java
package com.elbuensabor.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "detalleFactura")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class DetalleFactura extends Base {
    @Column
    private Integer cantidad;
    @Column
    private Double subtotal;

    @ManyToOne
    @JoinColumn(name = "factura_id")
    private Factura factura;

    @ManyToOne
    @JoinColumn(name = "articuloManufacturado_id")
    private ArticuloManufacturado articuloManufacturado;

    @ManyToOne
    @JoinColumn(name = "articuloInsumo_id")
    private ArticuloInsumo articuloInsumo;

}
